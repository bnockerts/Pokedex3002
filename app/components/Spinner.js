import React, { Component, PropTypes } from 'react';
import {
  ActivityIndicator,
  StyleSheet,
  View
} from 'react-native';

class Spinner extends Component {
  static propTypes = {
    isLoading: PropTypes.bool.isRequired
  }

  render() {
    return (
      <View style={styles.spinner}>
        <ActivityIndicator 
          animating={this.props.isLoading}
          size='large'
        />
      </View>
    )
  }
}

const styles = StyleSheet.create({
  spinner: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center'
  }
});

export default Spinner;
