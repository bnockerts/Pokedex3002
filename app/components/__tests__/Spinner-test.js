import React from 'react';
import renderer from 'react-test-renderer';
import Spinner from '../Spinner';

it('renders a Spinner', () => {
  const component = renderer.create(
    <Spinner isLoading={true} />
  );

  expect(component).toMatchSnapshot();
});
