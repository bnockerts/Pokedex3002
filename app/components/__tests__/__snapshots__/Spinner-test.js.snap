exports[`test renders a Spinner 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }>
  <ActivityIndicator
    animating={true}
    color="#999999"
    hidesWhenStopped={true}
    size="large" />
</View>
`;
